Zend/zend_object_handlers.c

```diff
@@ -1189,6 +1189,10 @@ found:;
 		variable_ptr = &EG(error_zval);
 		goto exit;
 	}
+	/* value may have changed during initialization */
+	if (UNEXPECTED(Z_ISREF_P(value))) {
+		value = Z_REFVAL_P(value);
+	}
```

Comment:
This doesn't seem to be covered by the test.

Reply:
I also wonder if this should back-up `value` instead. I'd find this a bit surprising:

```php
class C {
    public $s;
}

print "# Ghost:\n";

$r = new ReflectionClass(C::class);

$val = 1;
$o = $r->newLazyGhost(function () {
    global $val;
    $val = 2;
});
$o->s = $val;
var_dump($o);

print "# Proxy:\n";

$val = 1;
$o = $r->newLazyProxy(function ($obj) {
    global $val;
    $val = 2;
    return new C();
});
$o->s = $val;
var_dump($o);
```

```
# Ghost:
object(C)#%d (1) {
  ["s"]=>
  int(2)
}
# Proxy:
lazy proxy object(C)#%d (1) {
  ["instance"]=>
  object(C)#%d (1) {
    ["s"]=>
    int(2)
  }
}
```

Reply:
> This doesn't seem to be covered by the test

Indeed, this case is was reached anymore due to the exception

> I also wonder if this should back-up value instead. I'd find this a bit surprising:

Good point. I've updated to back up the value